<!DOCTYPE html>
<html lang="zh-CN">
    <head>
        <meta charset="UTF-8" />
        <meta http-equiv="X-UA-Compatible" content="IE=edge" />
        <meta name="viewport" content="width=device-width, initial-scale=1.0" />
        <title>09_try_catch捕获异常</title>
    </head>
    <body>
        <p>123</p>
        <script>
            function fn() {
                try {
                    // 可能发生错误的代码，要写到try
                    const p = document.querySelector(".p");
                    p.style.color = "red";
                } catch (error) {
                    // 拦截错误，提示浏览器提供的错误信息，但是不中断程序的执行。
                    console.log(error.message);
                    // 需要加return，才能中断后续的执行
                    // return;
                    // 或者使用抛出异常
                    throw new Error("是不是选择器错误了！");
                } finally {
                    // 不管程序对不对，我都要执行！
                    alert("不管程序对不对，我都要执行！");
                }

                console.log("如果不return，还是会执行这个！");
            }
            fn();
        </script>
    </body>
</html>
